### DISCUSSION TITLE
Suggest packages of used required packages

### DISCUSSION THREAD

**Discussion by filipagh** [composer-unused/composer-unused#363] [Ideas]
we require `ext-intl`
but composer-unused mark it as unused (we do not use it directly)

however some packages suggest it for some functions (`guzzlehttp/guzzle`), or better performance (`symfony/polyfill-intl-idn`)

should we add it to ignore, 

or if is suggested can composer-unused count it as used?

or create separate category for used suggested packages?

**Comment by icanhazstring** [comment]
This could be, that this might be used by a dependency you are using.
Which points down to #235.

So as you mentioned, that guzzle is having an implementation to check if certain functions from `ext-intl` are present, and then uses them, otherwise, it won't.

**Reply by icanhazstring** [reply]
We could, ofc also just parse the `suggest` section of a `composer.json` from a dependency and mark it the same way as `required-by` does. But I think #235 would be the more solid solution ;)
